joomla/joomla-cms
# [/cms/component/helper] Exclude components in discover state from list

### Summary of Changes

Exclude components in discover state from list.

Method "getComponents" in class JComponentHelper return all components, even those who are in the state to discover.  Must add state in queries.
### Test instructions
1. In joomla components dir, create a directory named "com_test" for example,
2. In this dir, create one file named "router.php",
3. Edit this file and paste this code :
   `<?php 
   defined('_JEXEC') or die;
   die('Router call');
   ?>`
4. In this dir, create one file named "test.xml",
5. Edit this file and paste this :
   `<?xml version="1.0" encoding="utf-8"?>
   <extension type="component" version="3.1" method="upgrade">
   <name>com_test</name>
   <files folder="site">
       <filename>router.php</filename>
   </files>
   </extension>
   ` 
6. In backend, go to "Extensions => Manage => Discover"
7. Click "Discover" button, you must see "com_test" in list,
8. Go to fronted, you must see message "Router call",
9. Apply patch,
10. Go to fronted, no message, router not loaded
### Related pull request

**#8986 Parse preprocess rules from component routers :** router of all components are loaded, even those that are not installed.
